osbenchmark/builder/executors/shell_executor.py

```diff
@@ -0,0 +1,19 @@
+import shutil
+import subprocess
+
+from osbenchmark.builder.executors.executor import Executor
+from osbenchmark.exceptions import ExecutorError
+from osbenchmark.utils import process
+
+
+class ShellExecutor(Executor):
```

Comment:
Spoke offline: To conform to the [RFC's class diagram](https://user-images.githubusercontent.com/62891993/155002622-58b549d0-fd6a-43de-91d2-fe6b63c6e69d.png), it would be more clear to use ```LocalExecutor```.

Reply:
Might also be worth adding the RFC's class diagram to the description and pointing PR reviewers to it to give them a better idea of what these changes will later amount to.